File: lc-1310.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        int[] pre = new int[n];
        int base = 0;
        for (int i = 0; i < arr.length; i++) {
            base = base ^ arr[i];
            pre[i] = base;
        }
        int[] res = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int tmp = pre[queries[i][1]];
            if (queries[i][0] != 0) {
                tmp = tmp ^ pre[queries[i][0] - 1];
            }
            res[i] = tmp;
        }
        return res;
    }
}

File: lc-1310.go
package main

func xorQueries(arr []int, queries [][]int) []int {
	xors := make([]int, len(arr))
	base := 0
	for i := 0; i < len(arr); i++ {
		base = base ^ arr[i]
		xors[i] = base
	}
	res := make([]int, len(queries))
	for i, query := range queries {
		tmp := xors[query[1]]
		if query[0] != 0 {
			tmp = tmp ^ xors[query[0]-1]
		}
		res[i] = tmp
	}
	return res
}
